degirum_tools/regression_eval.py

```diff
@@ -0,0 +1,111 @@
+#
+# regression_eval.py: evaluation toolkit for regression models used in PySDK samples
+#
+# Copyright DeGirum Corporation 2023
+# All rights reserved
+#
+
+import yaml
+import json
+import os
+import numpy as np
+
+
+class ImageRegressionModelEvaluator:
+    def __init__(
+        self,
+        dg_model,
+        input_resize_method="bilinear",
+        input_pad_method="crop-first",
+        image_backend="opencv",
+        input_img_fmt="JPEG",
+        input_letterbox_fill_color=(114, 114, 114),
+        input_numpy_colorspace="auto",
+    ):
+        """
+        Constructor.
+            This class evaluates the MAE and MSE for Image Regression models.
+
+            Args:
+                dg_model (Regression model): Regression model from the DeGirum model zoo.
+                input_resize_method (str): Input Resize Method.
+                input_pad_method (str): Input Pad Method.
+                image_backend (str): Image Backend.
+                input_img_fmt (str): InputImgFmt.
```

Comment:
Question: we do we need to pass all these model properties as function parameters just to assign them into the model object? If someone needs to change some of them, he can do it in his code, passing to your function already configured model. Or you want to enforce explicitly specifying all these model properties? If so, please provide clear comments, what they mean. Comment like `input_resize_method (str): Input Resize Method.` are pretty meaningless. You can just refer to the corresponding Model class attribute, like:

   input_resize_method (str): value to be assigned to model.input_resize_method property

Also lets rename `dg_model` to `model`

Reply:
Also, changing the model object just by passing it to some class constructor does not seem right.

Reply:
The reason behind this is to configure a model using the init_from_yaml method.